spot_rl_experiments/spot_rl/utils/geometry_utils.py

```diff
@@ -134,3 +137,117 @@ def is_pose_within_bounds(
     )
 
     return bool(is_within_linear_bounds and is_within_angular_bounds)
+
+
+def is_position_within_bounds(
+    test_position, target_position, xy_dist_threshold, z_dist_threshold
+) -> bool:
+    """
+    Check if a test position(x,y,z) is within  of target pose.
+
+    Args:
+        test_position (list): Test position, as [x,y,z]
+        target_position (list): Target position, as [x,y,z]
+        xy_dist_threshold (float): Threshold for xy distance (in meters)
+        z_dist_threshold (float): Threshold for z distance (in meters)
+
+    Returns:
+        is_within (bool): True if the test pose is within both linear and angular bounds of the target pose, False otherwise.
+    """
+    # Linear xy dist bounds
+    is_within_linear_xy_bounds = (
+        euclidean(test_position[:2], target_position[:2]) < xy_dist_threshold
+    )
+
+    # Linear z dist bounds
+    is_within_linear_z_bounds = (
+        abs(test_position[2] - target_position[2]) < z_dist_threshold
+    )
+
+    return bool(is_within_linear_xy_bounds and is_within_linear_z_bounds)
+
+
+def interpolation_between_two_positions(point1, point2, steps, kind="linear"):
```

Comment:
Beware that we might need to change this once we move to more constrained environments. Can you test the place controller with BD API in the home environment actually?

Reply:
We are actually not using the method `interpolation_between_two_positions()` right now.

Currently we are using `generate_intermediate_point()` to create an intermediate goal point and asking BD to move robot first to the intermediate goal point, and from there to the final goal.